web/source/osk/visualKeyboard.ts

```diff
@@ -679,6 +679,87 @@ namespace com.keyman.osk {
       }
     }
 
+    get width(): number {
+      if(this._width) {
+        return this._width;
+      } else {
+        return this.loadSizeFromCookie().width;
+      }
+    }
+
+    get height(): number {
+      if(this._height) {
+        return this._height;
+      } else {
+        return this.loadSizeFromCookie().height;
+      }
+    }
+
+    protected loadSizeFromCookie(): {width: number, height: number} {
+      let keyman = com.keyman.singleton;
+      let util = keyman.util;
+
+      var c = util.loadCookie('KeymanWeb_OnScreenKeyboard');
+      var newWidth: number, newHeight: number;
+      if(typeof(c) == 'undefined' || c == null) {
+        newWidth = screen.width * 0.3;
+        newHeight = 144; // 16 * 8; 16 is our preferred default font size.
+      } else {
+        // Restore OSK size - font size now fixed in relation to OSK height, unless overridden (in em) by keyboard
+        newWidth=util.toNumber(c['width'], 0.3 * screen.width); // Default - 30% of screen's width.
+
+        if(newWidth < 0.2*screen.width) {
+          newWidth = 0.2*screen.width;
+        } else if(newWidth > 0.9*screen.width) {
+          newWidth=0.9*screen.width;
+        }
+
+        newHeight=util.toNumber(c['height'],0.15*screen.height);
```

Comment:
The default height should be a factor of the width rather than of screen height -- and should be the same value for line 706 above...